File: kotlin-spring-boot-mybatis/src/main/java/com/xiao/boot/mybatis/testing/TestKtSpringDatabase.java
package com.xiao.boot.mybatis.testing;

import com.xiao.boot.mybatis.database.BaseDatabase;

import java.lang.annotation.ElementType;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Repeatable(TestKtSpringDatabases.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface TestKtSpringDatabase {
    Class<? extends BaseDatabase> database();
    Class[] mappers() default {};
}

File: kotlin-spring-boot-mybatis/src/main/kotlin/com/xiao/boot/mybatis/testing/TestKtSpringDatabase.kt
package com.xiao.boot.mybatis.testing

import com.xiao.boot.mybatis.database.BaseDatabase
import kotlin.reflect.KClass

/**
 *
 * @author lix wang
 */
@Repeatable
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class TestKtSpringDatabase(
    val database: KClass<out BaseDatabase>,
    val mappers: Array<KClass<*>> = []
)